x/gov/codec.proto

```diff
@@ -174,48 +160,35 @@ message Vote {
+message CreateProposalBase {
   // Human readable title. Must match `^[a-zA-Z0-9 _.-]{4,128}$`
-  string title = 2;
+  string title = 1;
   // Human readable description with 3 to 5000 chars.
-  string description = 3;
+  string description = 2;
   // ElectionRuleID is a reference to the election rule
-  bytes election_rule_id = 4 [(gogoproto.customname) = "ElectionRuleID"];
-  // ElectorateID is the reference to the electorate to define the group of possible voters.
-  bytes electorate_id = 5 [(gogoproto.customname) = "ElectorateID"];
+  bytes election_rule_id = 3 [(gogoproto.customname) = "ElectionRuleID"];
   // Unix timestamp when the proposal starts. Must be in the future.
-  int64 start_time = 6 [(gogoproto.casttype) = "github.com/iov-one/weave.UnixTime"];
+  int64 start_time = 4 [(gogoproto.casttype) = "github.com/iov-one/weave.UnixTime"];
   // Author is an optional field to set the address of the author with a proposal. The author must sign the message.
   // When not set it will default to the main signer.
-  bytes author = 7 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+  bytes author = 5 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
 }
 
-// CreateElectorateUpdateProposalMsg creates a new governance proposal to update an electorate.
-message CreateElectorateUpdateProposalMsg {
+// CreateProposalMsg creates a new governance proposal.
+// Metadata is from CreateProposalBase, actual proposal details from app-specific raw_option
+message CreateProposalMsg {
   weave.Metadata metadata = 1;
-  // Human readable title. Must match `^[a-zA-Z0-9 _.-]{4,128}$`
-  string title = 2;
-  // Human readable description with 3 to 5000 chars.
-  string description = 3;
-  // ElectorateID is the reference to the electorate that defines the group of possible voters.
-  bytes electorate_id = 4 [(gogoproto.customname) = "ElectorateID"];
-  // Unix timestamp when the proposal starts. Must be in the future.
-  int64 start_time = 5 [(gogoproto.casttype) = "github.com/iov-one/weave.UnixTime"];
-  // Author is an optional field to set the address of the author with a proposal. The author must sign the message.
-  // When not set it will default to the main signer.
-  bytes author = 6 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
-  // DiffElectors contains the changes that should be applied. Adding an address should have a positive weight, removing
-  // with weight=0.
-  repeated Elector diff_electors = 7 [(gogoproto.nullable) = false];
+  CreateProposalBase base = 2;
+  bytes raw_option = 3;
 }
 
 // DeleteProposalMsg deletes a governance proposal.
 message DeleteProposalMsg {
   weave.Metadata metadata = 1;
-  // ID is the unique identifier of the proposal to delete
-  bytes id = 2 [(gogoproto.customname) = "ID"];
+  // ProposalID is the unique identifier of the proposal to delete
+  bytes proposal_id = 2 [(gogoproto.customname) = "ProposalID"];
```

Comment:
:+1: I also prefer to avoid `ID` in favor of full name. The gain is not clear in this case, but once there is more than one "id" attribute it is straightforward if all are full names.